driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -35,15 +40,26 @@ struct Context {
 impl UpdatingOrderbook {
     /// Does not block on initializing the orderbook. This will happen in the first call to
     /// `get_auction_data` which can thus take a long time to complete.
-    pub fn new(contract: Arc<dyn StableXContract + Send + Sync>, web3: Web3) -> Self {
+    pub fn new(
+        contract: Arc<dyn StableXContract + Send + Sync>,
+        web3: Web3,
+        path: Option<PathBuf>,
+    ) -> Self {
+        // Recover the orderbook from file (if possible, otherwise use default).
+        let orderbook = match &path {
+            Some(path) => Orderbook::try_from(path.as_path()).unwrap_or_default(),
```

Comment:
Maybe we should log a warning or something if this fails but was expected to succeed?

Reply:
We don't know whether this is suspected to succeed. The first time you run you might not have a cache yet but still pass the path in order to create it. I don't mind the warning though as a way we can see that it works in the logs.

Reply:
Just spitballing but if `path` doesn't exist then `Ok(Orderbook::default())` can be returned instead of an error? That was we don't get false-positive warning logs when there is no cache.

Reply:
So, if I understand the discussion here, we would like to include a warning that file was not found before using default and logging something different when we use default orderbook because there was no path? Or simply just log some warning that the orderbook at specified path was not there?

Reply:
I have made the suggested changes here. Would be nice if I could get one last review of this particular chunk of code before merge! @e00E  and/or @nlordell